### DISCUSSION TITLE
Execute command to retrieve database password on connection

### DISCUSSION THREAD

**Discussion by danielbecroft** [dbeaver/dbeaver#36536] [General]
We are currently running PostgreSQL databases inside of Azure, and are using SSO (so we login to the database(s) using our normal AD/O365 accounts). This is all working fine.

However, in order to login, we need to get an access token from the azure cli, something like:

```
az account get-access-token --resource-type oss-rdbms
```
which returns a response in the format of:
```
{
  "accessToken": "TOKEN",
  "expiresOn": "...",
  "subscription": "...",
  "tenant": "...",
  "tokenType": "Bearer"
}
```

I've got this configured to run via Powershell and copy the password to the clipboard, so on connection I can run this and connect.

It would be good if there was a way to configure the connection password as a "shell script", or "shell command". So I could open dbeaver, hit Connect, the configuration says "run this to get the password", and the script is executed, token retrieved, and the login continues. 

Alternatively, some way to run this command outside of dbeaver, and update the stored database connection configuration with this new token so it's available automatically inside of dbeaver.

Is this something that's already possible, or something that could be added to the idea list?

**Comment by HocKu7** [comment]
Thank you for the idea.

**Comment by E1izabeth** [comment]
We support this type of authentification in our commercial products. I suggest taking a free two-week trial at https://dbeaver.com/trial/. For the complete list of advantages of our commercial versions and a comparison list, please check out this link: https://dbeaver.com/edition/

**Comment by E1izabeth** [comment]
Here is our wiki about it https://github.com/dbeaver/dbeaver/wiki/Azure-Cloud-Explorer

**Comment by serge-rider** [comment]
We do not provide autoatic way to retrieve a password or token becuase it is very unsecure. Anybody who has even short-term access to end-user machine could retrieve saved password in a second.

In PRO products dbeaver has project-based security - workspace or a project can be protected by master password and then you can retrieve database credentials.
